### PR TITLE
add the ability to delete versions via the API

### LINKED ISSUE #19784: Add the ability to delete versions through the API
### Describe the problem and steps to reproduce it:
Developers have the option to disable and delete versions in DevHub. We should make these actions available in the [version edit](https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#version-edit) API as well.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #19784

### CODE REVIEW

File: src/olympia/addons/tests/test_views.py [hunk 1]
```diff
@@ -3940,6 +3940,64 @@ class TestVersionViewSetUpdateJWTAuth(TestVersionViewSetUpdate):
...
+            api_version='v5',
+        )
+
+    def test_delete(self):
+        assert not self.version.deleted
+
+        response = self.client.delete(self.url)
+        assert response.status_code == 401
+        assert not self.version.reload().deleted
+
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/addons/tests/test_views.py:3963]: Also... a regular reviewer (without `Addons:Edit` at least) shouldn't be allowed to delete versions. As this stands it looks like they would ?

[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/addons/tests/test_views.py:3963]: https://github.com/mozilla/addons-server/pull/20143/commits/66ac375cec537ba592f6e26c55c62cf00cd5fc32

File: src/olympia/addons/tests/test_views.py [hunk 2]
```diff
@@ -3940,6 +3940,84 @@ class TestVersionViewSetUpdateJWTAuth(TestVersionViewSetUpdate):
...
+    def test_not_author_cannot_delete(self):
+        another_user = user_factory(read_dev_agreement=self.days_ago(0))
+        self.client.login_api(another_user)
+        response = self.client.delete(self.url)
+        assert response.status_code == 403
+        assert not self.version.reload().deleted
+
+        # even if they are a reviewer
+        self.grant_permission(another_user, ':'.join(amo.permissions.ADDONS_EDIT))
```
Comment (diox) [inline src/olympia/addons/tests/test_views.py:3978]: It's nice to test `ADDONS_EDIT` but we should also test with something that satisfies `acl.is_reviewer()`

**Review by diox — APPROVED** [review]

File: src/olympia/addons/tests/test_views.py (hunk 2 above)
Comment (eviljeff) [reply src/olympia/addons/tests/test_views.py]: https://github.com/mozilla/addons-server/pull/20143/commits/9d099dd16899b0d939c7430db89e91095133fadc

**Pull request merged**